holochain.go

```diff
@@ -671,8 +685,48 @@ func (h *Holochain) HashSpec() HashSpec {
 	return h.hashSpec
 }
 
+// SendAsync builds a message and either delivers it locally or over the network via node.Send but registers a function for asyncronous call back
+func (h *Holochain) SendAsync(proto int, to peer.ID, t MsgType, body interface{}, callback *Callback, timeout time.Duration) (err error) {
+	var response interface{}
+
+	// TODO add timeout
+	go func() {
+		response, err = h.Send(context.Background(), proto, to, t, body, timeout)
+		if err == nil {
+			var r Ribosome
+			r, _, err := h.MakeRibosome(callback.zomeType)
+			if err != nil {
+				h.asyncSends <- err
+				return
+			}
+			//var result interface{}
+			_, err = r.RunAsyncSendResponse(response, callback.Function, callback.ID)
```

Comment:
I really want SendAsync to be the general function for asynchronous sending, and I do think it will neeed to handle results from other protocols, so for now I converted the call to  RunAsyncSendResponse to take an AppMsg and put the type switch in AsyncSend with an "unimplemented" error  for other types.